### PR TITLE
Removes 'Tag' and moves its utility methods to a 'StringSanitization' class.

### PR DESCRIPTION
**Pull request by sebright**

### CODE REVIEW

**Comment by sebright** [comment]
@dinooliva I resolved the conflicts.

File: core/java/com/google/census/StringSanitization.java [hunk 1]
```diff
@@ -13,52 +13,10 @@
...
-  public int hashCode() {
-    return key.hashCode() * 31 + value.hashCode();
-  }
-
-  // Assumes key and value have already been sanitized.
-  Tag(String key, String value) {
-    this.key = key;
-    this.value = value;
-  }
+final class StringSanitization {
```
Comment (dinooliva) [inline core/java/com/google/census/StringSanitization.java]: Consider naming the class something a little more generic like TagUtil or TagStringUtil.

Comment (sebright) [inline core/java/com/google/census/StringSanitization.java]: I renamed it to StringUtil, since it also handles MetricNames.

**Comment by sebright** [comment]
Thanks!

**Pull request merged**